ant/src/initialize.lua

```diff
@@ -0,0 +1,26 @@
+local json = require(".json")
+local utils = require(".utils")
+
+local initialize = {}
+
+function initialize.initialize(msg)
+	local state = msg.Data and json.decode(msg.Data)
+	if not state then
+		return utils.reply("State not provided")
+	end
+
+	local name, ticker, balances, controllers, records =
+		state.name, state.ticker, state.balances, state.controllers, state.records
+
+	if not name or not ticker or not balances or not controllers or not records then
```

Comment:
i would do some more assertions. an example of how we assert properties of a table in the contract 

https://github.com/ar-io/ao-pilot/blob/develop/contract/src/arns.lua#L231-L249